src/auslib/util/versions.py

```diff
@@ -43,10 +43,49 @@ class AncientMozillaVersion(StrictVersion):
     )
 
 
+class GlobVersionTuple(tuple):
+    def __eq__(self, value):
+        for i, j in enumerate(self):
+            if j != value[i]:
+                return False
+        return True
+
+    def __ne__(self, value):
+        return not self.__eq__(value)
+
+
+class GlobVersion(StrictVersion):
+    """A version class that supports Firefox versions 5.0 and up, which ends
+    with a glob `*`. Not really a StrictVersion at all, but it needs to be
+    to compare with other MozillaVersions."""
+
+    version_re = re.compile(
+        r"""^(\d+) \. (\d+\.\*|\*)$""",
```

Comment:
Feel free to shoot me done if you think this is YAGNI, but I'm surprised we're only supporting globbing two part versions when we could easily support three part versions as well.

Reply:
The 2nd parens is either a minor version + `.*` for the mini version, or just a `.*` for the minor version.
We don't use the regex groups in the `GlobVersion.parse` function; we just use the regex to make sure the schema of the version string is valid.

Reply:
Right, what I'm saying is that 89.89.5* wouldn't work, for example. I can live with this, it just feels like a weird limitation.

Reply:
I think we covered this [here](https://github.com/mozilla-releng/balrog/pull/1928#discussion_r614832974). `89.89.5*` would match `89.89.5`, `89.89.51`, and `89.89.509`; that seems more like a footgun than a feature.

Reply:
ok, ty